spec/index.bs

```diff
@@ -1950,6 +1975,17 @@ interface GPUAdapterInfo {
         instead set |adapterInfo|.{{GPUAdapterInfo/description}} to the empty string or a
         reasonable approximation of a description.
 
+    1. For each supported {{GPUFeatureName|feature}} with associated {{GPUAdapterProperties|properties}},
+        set the appropriate values:
+        * If {{GPUFeatureName/"subgroups"}} is supported, set:
+            * {{GPUAdapterProperties/minSubgroupSize}} to the smallest supported subgroup size.
+            * {{GPUAdapterProperties/maxSubgroupSize}} to the largest supported subgroup size.
+
+        To preserve privacy, the user agent may choose to not support some features or provide values
+        for the properties which do not distinguish different devices, but are still usable
+        (e.g. set {{GPUAdapterProperties/minSubgroupSize}} to 4 and {{GPUAdapterProperties/maxSubgroupSize}}
+        to 128).
```

Comment:
Probably worth spelling out somewhere (in WGSL or API spec) that the  actual subgroup size is determined by the implementation and may depend on the shader code.
In other words the developer doesn't pick the subgroup size.  

In the API side 'minSubgroupSize' is a promise that the effective/actual subgroup size observable in a shader is at least this size. (And similar for max).
And this is why user agents can relax the reported properties in this way.

Reply:
It is mentioned in the WGSL spec in the subgroups section when describing subgroup size that the size is function of device properties and device compiler.